Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -0,0 +1,230 @@
+
+            if (EntityManager.TryGetComponent<SharedButcherableComponent>(args.VictimUid, out var butcherable))
+                butcherable.BeingButchered = false;
+
+            if (Spikeable(uid, args.UserUid, args.VictimUid, component, butcherable))
+            {
+                Spike(uid, args.UserUid, args.VictimUid, component);
+            }
+        }
+
+        private void OnDragDrop(EntityUid uid, KitchenSpikeComponent component, DragDropEvent args)
+        {
+            if(args.Handled)
+                return;
+
+            if (!Spikeable(uid, args.User, args.Dragged, component))
+                return;
+
+            if (TrySpike(uid, args.User, args.Dragged, component))
+                args.Handled = true;
+        }
+
+        private void OnInteractUsing(EntityUid uid, KitchenSpikeComponent component, InteractUsingEvent args)
+        {
+            if (args.Handled)
+                return;
+            
+            if (TryGetPiece(uid, args.User, args.Used))
+                args.Handled = true;
+        }
+
+        private void Spike(EntityUid uid, EntityUid userUid, EntityUid victimUid,
+            KitchenSpikeComponent? component = null, SharedButcherableComponent? butcherable = null)
+        {
+            if (!Resolve(uid, ref component) || !Resolve(victimUid, ref butcherable))
+                return;
+
+            component.MeatPrototype = butcherable.MeatPrototype;
+            component.MeatParts = butcherable.Pieces;
+
+            // This feels not okay, but entity is getting deleted on "Spike", for now...
+            component.MeatSource1p = Loc.GetString("comp-kitchen-spike-remove-meat", ("victim", victimUid));
+            component.MeatSource0 = Loc.GetString("comp-kitchen-spike-remove-meat-last", ("victim", victimUid));
+            component.MeatName = Loc.GetString("comp-kitchen-spike-meat-name", ("victim", victimUid));
+
+            UpdateAppearance(uid, null, component);
+
+            _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-kill", ("user", userUid), ("victim", victimUid)), uid, Filter.Pvs(userUid));
+
+            // THE WHAT?
+            // TODO: Need to be able to leave them on the spike to do DoT, see ss13.
+            EntityManager.QueueDeleteEntity(victimUid);
+
+            SoundSystem.Play(Filter.Pvs(uid), component.SpikeSound.GetSound(), uid);
+        }
+
+        private bool TryGetPiece(EntityUid uid, EntityUid user, EntityUid used,
+            KitchenSpikeComponent? component = null, UtensilComponent? utensil = null)
+        {
+            if (!Resolve(uid, ref component) || component.MeatParts == 0)
+                return false;
+
+            // Is using knife
+            if (!Resolve(used, ref utensil, false) || (utensil.Types & UtensilType.Knife) == 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-knife-needed"), uid, Filter.Entities(user));
+                return false;
+            }
```

Comment:
So this is something that I care about in general, but is maybe not so important. In this specific case, given that something with `KitchenSpikeComponent` probably has no other systems/components with interactions, but:

Given that you only mark the event as handled if TryGetPiece returns true, you should either always mark the event as handled regardless of the return value make TryGetPiece return true if **any** interaction occurred, including popup messages, and not just on success.

If you don't do that in general, then you get instances where a single interaction can generate multiple simultaneous pop-up messages. For example, this previously happened with locked lockers. If you tried to open one without access, it would simultaneously generate both an "Access Denied" (from lock system) and a "This is locked" (from entity storage).

Reply:
Is there any priority for Events Subscription (Handle it last)?

Reply:
You can specify that your subscription should go before or after other systems. e.g., anchorable (wrench interactions) has:
```cs
SubscribeLocalEvent<AnchorableComponent, InteractUsingEvent>(OnInteractUsing, after:new []{typeof(ConstructionSystem)});
```

IIRC if you specify some after/before ordering it is apparently slower or something so you're supposed to try avoid it if possible. Not sure how important that actually is.

Reply:
And you might actually have to do something with anchorable.

Currently if you try to wrench something it generates a "You need a knife!" popup while just un-anchoring it like normal. While maybe a confusing "error" message, at least that interaction still works. If you just marked the event as handled, it would just generate the pop-up and then prevent you from actually un-anchoring it while there is a meat on the spike.

Reply:
damn, "after" would be great, but what I can do is popup on using empty InteractHandEvent and handle it (guess there will be no other components with this subscription on simple thing like spike), so InteractUsing stays Handled only on success and un-anchoring works fine (but no PopUp on using wrong item), or MAYBE alt + click for butcher (this is gonna be confusing)